src/frequenz/sdk/timeseries/_resampling.py

```diff
@@ -495,8 +495,13 @@ async def resample(self, *, one_shot: bool = False) -> None:
                     self._config.resampling_period,
                 )
 
+            # We need to make a copy here because we need to match the results to the
+            # current resamplers, and since we await here, new resamplers could be added
+            # or removed from the dict while we awaiting the resampling, which would
+            # cause the results to be out of sync.
+            resamplers = self._resamplers.copy()
```

Comment:
Right, right, right. I got confused because we are using the `values()` right after the copy, but when we access those `values()` there was no `await` yet, so we can use the original `dict` there. :+1: